integration/test/validation/app_test.go

```diff
@@ -0,0 +1,65 @@
+// +build k8srequired
+
+package validation
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestFailWhenCatalogNotFound(t *testing.T) {
+	ctx := context.Background()
+
+	var err error
+
+	app := &v1alpha1.App{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "dex-app-unique",
+			Namespace: "giantswarm",
+			Labels: map[string]string{
+				label.AppOperatorVersion: "0.0.0",
+			},
+		},
+		Spec: v1alpha1.AppSpec{
+			Catalog:   "missing",
+			Name:      "dex-app",
+			Namespace: "giantswarm",
+			KubeConfig: v1alpha1.AppSpecKubeConfig{
+				InCluster: true,
+			},
+			Version: "1.2.2",
+		},
+	}
+	expectedError := "validation error: catalog `missing` not found"
+
+	logger.LogCtx(ctx, "level", "debug", "message", "waiting for failed app creation")
+
+	o := func() error {
+		err = appTest.CtrlClient().Create(ctx, app)
```

Comment:
@fiunchinho I'd like to release a new version of apptest but what should we do about https://github.com/giantswarm/azure-admission-controller/pull/66#discussion_r517860678

Should we remove the `.CtrlClient()` wrapper and wire it separately in the tests?

Reply:
I spoke to @tuommaki and IIUC the issue with having a shared `CtrlClient()` it not the client itself, but the one k8s API behind. Several different tests will share the same state, since there is only one k8s API. So the problem doesn't go away making it harder to get the client instance. I think it's a problem with integration tests per se, and people should be aware of it.

So I'd keep the `CtrlClient()` method and use `appTest` from a global var.

Reply:
OK cool thanks for the explanation. I'll leave the `CtrilClient` in place then.

Yes shared state within a single test is an issue. We can have separate tests but then we need a Circle VM per test and there can be cloud provider costs.